app/admin-tab/machine/controller.js

```diff
@@ -0,0 +1,128 @@
+import Ember from 'ember';
+import Sortable from 'ui/mixins/sortable';
+import C from 'ui/utils/constants';
+
+export default Ember.Controller.extend(Sortable, {
+  application: Ember.inject.controller(),
+  settings:    Ember.inject.service(),
+  newMachine: null,
+  sortBy: 'name',
+  sorts: {name: ['name']},
+  upgrading: false,
+
+  actions: {
+    activate: function(driver) {
+      let action = null;
+      if (driver.get('actionLinks.activate')) {
+        action = 'activate';
+      } else if (driver.get('actionLinks.reactivate')) {
+        action = 'reactivate';
+      }
+
+      driver.doAction(action);
+    },
+    addNewDriver: function(driver) {
+      let newDriver = {
+        type            : 'machineDriver',
+        name            : null,
+        description     : null,
+        checksum        : null,
+        url             : null,
+        activateOnCreate: true,
+      };
+
+      if (driver) {
+        newDriver.name        = driver.name;
+        newDriver.description = driver.description;
+        newDriver.checksum    = driver.checksum;
+        newDriver.url         = driver.url;
+        newDriver.externalId  = driver.id;
+      }
+
+      this.set('newMachine', this.get('userStore').createRecord(newDriver));
+
+      this.get('application').setProperties({
+        editMachineDriver: true,
+        originalModel: this.get('newMachine'),
+      });
+    },
+    addCatalogDriver: function(driver) {
+      this.get('store').request({url: this.get('app.catalogEndpoint')+'/templates/'+driver.id}).then((template) =>{
+        this.get('store').request({url: template.versionLinks[template.defaultVersion]}).then((driver) =>{
+          let newDriver = {
+            type            : 'machineDriver',
+            description     : (driver.description || null),
+            checksum        : (driver.files.checksum || null),
+            url             : driver.files.url.replace(/[\n\r]+/g, ''),
+            externalId      : driver.id,
+            activateOnCreate: true,
+          };
+          this.get('userStore').createRecord(newDriver).save().then((newDriver) => {
+            this.get('model.drivers').pushObject(newDriver);
+          });
+        });
+      });
```

Comment:
Add a catch() and show a growl in case the catalog API fails (inject growl and then `growl.fromError(err)`)